Tests/WorkspaceTests/InitTests.swift

```diff
@@ -14,398 +14,470 @@ import Basics
+    ///   - name: The name of the package (defaults to "Foo")
+    ///   - supportedTestingLibraries: The testing libraries to include
+    ///   - buildSystem: The build system to use for verification
+    ///   - customVerification: Additional verification steps to perform
+    /// - Returns: The path to the created package
+    private func createAndVerifyPackage(
+        packageType: InitPackage.PackageType,
+        name: String = "Foo",
+        supportedTestingLibraries: Set<TestingLibrary> = [.xctest],
+        buildSystem: BuildSystemProvider.Kind? = nil,
+        customVerification: ((AbsolutePath, String) throws -> Void)? = nil
+    ) async throws {
+        return try await testWithTemporaryDirectory { tmpPath in
             let fs = localFileSystem
-            let path = tmpPath.appending("Foo")
-            let name = path.basename
+            let path = tmpPath.appending(name)
+            let packageName = path.basename
             try fs.createDirectory(path)
 
             // Create the package
             let initPackage = try InitPackage(
-                name: name,
-                packageType: .library,
-                supportedTestingLibraries: [.xctest],
+                name: packageName,
+                packageType: packageType,
+                supportedTestingLibraries: supportedTestingLibraries,
                 destinationPath: path,
-                fileSystem: localFileSystem
+                fileSystem: fs
             )
+
+            // Capture progress messages
             var progressMessages = [String]()
             initPackage.progressReporter = { message in
                 progressMessages.append(message)
             }
+
+            // Write the package structure
             try initPackage.writePackageStructure()
 
-            // Not picky about the specific progress messages, just checking that we got some.
-            XCTAssertGreaterThan(progressMessages.count, 0)
+            // Verify progress messages were generated
+            #expect(progressMessages.count > 0, "Expected progress messages during package creation")
 
-            // Verify basic file system content that we expect in the package
+            // Verify basic file system content
             let manifest = path.appending("Package.swift")
-            XCTAssertFileExists(manifest)
-            let manifestContents: String = try localFileSystem.readFileContents(manifest)
+            expectFileExists(at: manifest)
+
+            let manifestContents: String = try fs.readFileContents(manifest)
             let version = InitPackage.newPackageToolsVersion
             let versionSpecifier = "\(version.major).\(version.minor)"
-            XCTAssertMatch(manifestContents, .prefix("// swift-tools-version:\(version < .v5_4 ? "" : " ")\(versionSpecifier)\n"))
+            expectMatch(manifestContents, .prefix("// swift-tools-version:\(version < .v5_4 ? "" : " ")\(versionSpecifier)\n"))
```

Comment:
suggestion: can we use the standard API call instead of this helper as the latter may not provide the uses failure string? 

convert this to `#expect(manifestContents.hasPrefix("// swift-tools-version:\(version < .v5_4 ? "" : " ")\(versionSpecifier)\n")`

This comment applies to all other instances of `expectMatch` and `expectNoMatch`.